Apps/W1/EDocumentConnectors/FORNAV/app/src/ForNAV/AppRespHandler.Codeunit.al

```diff
@@ -0,0 +1,64 @@
+namespace Microsoft.EServices.EDocumentConnector.ForNAV;
+
+using System.Threading;
+using Microsoft.EServices.EDocument;
+
+
+/// <summary>
+/// Processes the FORNAV Incoming Documents
+/// </summary>
+codeunit 6410 "ForNAV App. Resp. Handler"
+{
+    Permissions =
+        tabledata Microsoft.eServices.EDocument."E-Document Service Status" = RIMD;
+    Access = internal;
+    TableNo = "Job Queue Entry";
+
+    local procedure UpdateServiceStatus(EDocument: Record "E-Document"; EDocumentService: Record "E-Document Service"; EDocumentStatus: Enum "E-Document Service Status")
+    var
+        EDocumentServiceStatus: Record "E-Document Service Status";
+    begin
+        EDocument.Get(EDocument."Entry No");
+        if EDocumentServiceStatus.Get(EDocument."Entry No", EDocumentService.Code) then begin
+            EDocumentServiceStatus.Validate(Status, EDocumentStatus);
+            EDocumentServiceStatus.Modify()
+        end else begin
+            EDocumentServiceStatus.Validate("E-Document Entry No", EDocument."Entry No");
+            EDocumentServiceStatus.Validate("E-Document Service Code", EDocumentService.Code);
+            EDocumentServiceStatus.Validate(Status, EDocumentStatus);
+            EDocumentServiceStatus.Insert();
+        end;
+    end;
+
+    procedure ProcessApplicationResponse(DocumentType: Enum "E-Document Type"; DocNo: Text; Status: Enum "E-Document Service Status"): Boolean
+    var
+        EDocument: Record "E-Document";
+        EDocumentService: Record "E-Document Service";
+        EDocLogHelper: Codeunit "E-Document Log Helper";
+    begin
+        if not EDocumentService.Get('FORNAV') then
```

Comment:
This design wont work. Ther is no guarantee for the code being this value. Since it is user facing, it bad idea to rely on it as well.